File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        seenNum = set()

        for num in nums:
            if num in seenNum:
                return True
            seenNum.add(num)

        return False

"""
1,2,3,3,4,4,5 - True
1,2,3,4 -> False
1,1,2,3,4 -> True
"""

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] products = new int[nums.length];
        products[0] = 1;
        int currProduct = nums[0];

        for(int i = 1; i < nums.length; i++) {
            products[i] = currProduct;
            currProduct *= nums[i];
        }

        currProduct = nums[nums.length - 1];
        for(int i = nums.length - 2; i >= 0; i--) {
            products[i] *= currProduct;
            currProduct *= nums[i];
        }

        return products;
    }
}  
